batch7crud-roll6/src/main/java/com/lftechnology/batch7crud/controller/UserController.java

```diff
@@ -0,0 +1,214 @@
+                logger.log(Level.SEVERE, e.getMessage());
+            } catch (IOException e) {
+                logger.log(Level.SEVERE, e.getMessage());
+            }
+        } else {
+            String[] pathArgs = urlString.split("/");
+            if ("".equals(pathArgs[0]) && "add".equals(pathArgs[1])) {
+                try {
+                    createProcess(request, response);
+                } catch (ServletException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                } catch (IOException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                }
+            } else if ("".equals(pathArgs[0]) && "edit".equals(pathArgs[2])) {
+                try {
+                    int userID = TypeCaster.toInt(pathArgs[1]);
+                    editProcess(request, response, userID);
+                } catch (ServletException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                } catch (IOException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                }
+            } else if ("".equals(pathArgs[0]) && "delete".equals(pathArgs[2])) {
+                try {
+                    int userID = TypeCaster.toInt(pathArgs[1]);
+                    deleteProcess(request, response, userID);
+                } catch (ServletException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                } catch (IOException e) {
+                    logger.log(Level.SEVERE, e.getMessage());
+                }
+            }
+
+        }
+
+    }
+
+    private void list(HttpServletRequest request, HttpServletResponse response, int page) throws ServletException, IOException {
+        try {
+            request.setAttribute("users", userService.fetch(page));
+            request.getRequestDispatcher("/WEB-INF/views/listUser.jsp").forward(request, response);
+
+        } catch (DataException e) {
+            logger.log(Level.SEVERE, e.getMessage());
+        }
+    }
+
+    private void create(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        request.getRequestDispatcher("/WEB-INF/views/addUser.jsp").forward(request, response);
+    }
+
+    private void createProcess(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        User user = new User();
+
+        String firstName = request.getParameter("firstname");
+        String surName = request.getParameter("surname");
+        String username = request.getParameter("username");
+        String password = request.getParameter("password");
+
+        user.setFirstName(firstName);
+        user.setSurName(surName);
+        user.setUserName(username);
+        user.setPassword(password);
+
+        try {
+            userService.addUser(user);
+            response.sendRedirect(REDIRECT_STRING);
+        } catch (DataException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)